core/controlplane/config/templates/cluster.yaml

```diff
@@ -1071,6 +1071,43 @@ addons:
 #   # When enabled, `kubectl drain` is run on the node going to shut-down
 #   nodeDrainer:
 #     enabled: true
+#   # Enable dex  integration - https://github.com/coreos/dex
+#   # Configure OpenID Connect token authenticator plugin in Kubernetes API server. Notice: always use "https" for the "url",
+#   # otherwise the Kubernetes API server will not start correctly. The "caFile" can be skipped if you use a certificate signed
+#   # by a trusted CA.
+#   dex:
+#     enabled: true
+#     url: "https://dex.example.com"
```

Comment:
Should we ideally provision an ELB and a record set for serving the url?
// However, it might be a separate task. I'd just like to discuss about this for now